src/tensorlake/function_executor/service.py

```diff
@@ -256,3 +284,165 @@ def get_info(
         self, request: InfoRequest, context: grpc.ServicerContext
     ) -> InfoResponse:
         return InfoResponse(**info_response_kv_args())
+
+    def _execute_task_in_thread(
+        self, run_request: RunTaskRequest, task_execution: _TaskExecution
+    ):
+        response = RunTaskResponse(
+            task_id=run_request.task_id,
+            outcome_code=TaskOutcomeCode.TASK_OUTCOME_CODE_FAILURE,
+            failure_reason=TaskFailureReason.TASK_FAILURE_REASON_INTERNAL_ERROR,
+            stderr="",
+            stdout="",
+        )
+
+        try:
+
+            def progress_reporter(progress: Progress):
+                with task_execution.updated:
+                    task_execution.progress = ProgressUpdate(
+                        current=progress.current, total=progress.total
+                    )
+                    task_execution.updated.notify_all()
+                # sleep(0) here momentarily releases the GIL, giving
+                # other threads a chance to run.
+                time.sleep(0)
+
+                # TODO: Consider implementing task cancellation here.
+
+            # Run the task using existing handler
+            response = RunTaskHandler(
+                request=run_request,
+                invocation_state=ProxiedInvocationState(
+                    run_request.task_id, self._invocation_state_proxy_server
+                ),
+                function_wrapper=self._function_wrapper,
+                function_stdout=self._function_stdout,
+                function_stderr=self._function_stderr,
+                graph_metadata=self._graph_metadata,
+                logger=self._logger,
+                progress_reporter=progress_reporter,
+            ).run()
+
+            if response.outcome_code == TaskOutcomeCode.TASK_OUTCOME_CODE_UNKNOWN:
+                response.outcome_code = TaskOutcomeCode.TASK_OUTCOME_CODE_FAILURE
+                response.failure_reason = (
+                    TaskFailureReason.TASK_FAILURE_REASON_INTERNAL_ERROR
+                )
+
+        except Exception as e:
+            # Handle any errors by creating a failed task result
+            self._logger.error(
+                "task execution failed in background thread",
+                task_id=run_request.task_id,
+                error=str(e),
```

Comment:
exc_info=e.

This enables exception formatting configured for the logger. exc_info is used in all code paths.

Reply:
Done (good catch, dunno what I was thinking there :-)